File: src/main/java/tech/jhipster/lite/shared/github/domain/GithubOrganization.java
package tech.jhipster.lite.shared.github.domain;

import tech.jhipster.lite.shared.error.domain.Assert;

public record GithubOrganization(String login, String name, String avatarUrl) {
  public GithubOrganization {
    Assert.notBlank("login", login);
    Assert.notBlank("name", name);
  }
}

File: src/main/webapp/app/module/domain/GithubOrganization.ts
type GithubOrganizationLogin = string;
type GithubOrganizationName = string;
type GithubOrganizationAvatarUrl = string;

export type GithubOrganization = {
  login: GithubOrganizationLogin;
  name: GithubOrganizationName;
  avatarUrl: GithubOrganizationAvatarUrl;
}
